lib/react/rails/engine.rb

```diff
@@ -3,7 +3,12 @@ module Rails
     class Engine < ::Rails::Engine
       initializer "react_rails.setup_engine", :group => :all do |app|
         sprockets_env = app.assets || Sprockets # Sprockets 3.x expects this in a different place
-        sprockets_env.register_engine(".jsx", React::JSX::Template)
+        if Gem::Version.new(Sprockets::VERSION) >= Gem::Version.new("3.0.0")
+          sprockets_env.register_mime_type("application/jsx", extensions: [".jsx", ".js.jsx"])
```

Comment:
Is this for handling requests with mime-type `application/jsx` ?  Where do those come from? 

For react-rails, I think all requests are `application/javascript`. If that's correct, let's remove this line since we don't need it!